.github/workflows/performance_harness_run.yaml

```diff
@@ -12,10 +25,131 @@ defaults:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: ${{fromJSON(needs.d.outputs.missing-platforms)}}
+    runs-on: ["self-hosted", "enf-x86-beefy"]
+    permissions:
+      packages: write
+      contents: read
+    steps:
+      - name: Login to Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{github.repository_owner}}
+          password: ${{secrets.GITHUB_TOKEN}}
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          tags: ${{fromJSON(needs.d.outputs.p)[matrix.platform].image}}
+          file: ${{fromJSON(needs.d.outputs.p)[matrix.platform].dockerfile}}
+
+  Build:
+    needs: [v, d, build-platforms]
+    if: always() && needs.v.result == 'success' && needs.d.result == 'success' && (needs.build-platforms.result == 'success' || needs.build-platforms.result == 'skipped')
+    runs-on: ["self-hosted", "enf-x86-beefy"]
+    container: ${{fromJSON(needs.d.outputs.p)[github.event.inputs.platform-choice].image}}
+    steps:
+        - name: Checkout leap ref
+          id: checkout
+          uses: actions/checkout@v3
+          with:
+            ref: '${{needs.v.outputs.leap-ref}}'
+            submodules: recursive
+        - name: Build
+          id: build
+          run: |
+            # https://github.com/actions/runner/issues/2033
+            chown -R $(id -u):$(id -g) $PWD
+            cmake -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr -GNinja
+            cmake --build build
+            tar -pc --exclude "*.o" build | zstd --long -T0 -9 > build.tar.zst
+        - name: Upload builddir
+          uses: AntelopeIO/upload-artifact-large-chunks-action@v1
+          with:
+            name: ${{github.event.inputs.platform-choice}}-build
+            path: build.tar.zst
+
+  tests:
+    name: Tests
+    needs: [v, d, Build]
+    if: always() && needs.Build.result == 'success'
+    strategy:
+      fail-fast: false
     runs-on: ubuntu-latest
+    container:
+      image: ${{fromJSON(needs.d.outputs.p)[github.event.inputs.platform-choice].image}}
+      options: --security-opt seccomp=unconfined
     steps:
-      - name: Workflow Stub
+      - name: Download builddir
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{github.event.inputs.platform-choice}}-build
+      - name: Run Performance Test
+        run: |
+          # https://github.com/actions/runner/issues/2033  -- need this because of full version label test looking at git revs
+          chown -R $(id -u):$(id -g) $PWD
+          zstdcat build.tar.zst | tar x
+          cd build
+          ./tests/performance_tests/performance_test.py ${{needs.v.outputs.test-params}}
+      - name: Prepare results
+        id: prep-results
         run: |
-          echo "Workflow Stub"
+          # https://github.com/actions/runner/issues/2033
+          chown -R $(id -u):$(id -g) $PWD
```

Comment:
Similar to above -- is 2033 workaround needed here? Seems unlikely since there is just a `tar`

Reply:
Addressed: 6b4f7452093ca4